android/KMEA/app/src/main/java/com/tavultesoft/kmea/data/KeyboardController.java

```diff
@@ -63,27 +63,7 @@ public synchronized void initialize(Context context) {
           ObjectInputStream inputStream = new ObjectInputStream(new FileInputStream(keyboards_dat));
           ArrayList<HashMap<String, String>> dat_list = (ArrayList<HashMap<String, String>>) inputStream.readObject();
           inputStream.close();
-
-          for(HashMap<String, String> kbdMap : dat_list) {
-            boolean isNewKeyboard = kbdMap.containsKey(KeyboardPickerActivity.KMKEY_INTERNAL_NEW_KEYBOARD) &&
-              kbdMap.get(KeyboardPickerActivity.KMKEY_INTERNAL_NEW_KEYBOARD).equals(KeyboardPickerActivity.KMKEY_INTERNAL_NEW_KEYBOARD);
-
-            Keyboard k = new Keyboard(
-              kbdMap.get(KMManager.KMKey_PackageID),
-              kbdMap.get(KMManager.KMKey_KeyboardID),
-              kbdMap.get(KMManager.KMKey_KeyboardName),
-              kbdMap.get(KMManager.KMKey_LanguageID),
-              kbdMap.get(KMManager.KMKey_LanguageName),
-              MapCompat.getOrDefault(kbdMap, KMManager.KMKey_Version, "1.0"),
-              MapCompat.getOrDefault(kbdMap, KMManager.KMKey_CustomHelpLink, ""),
-              MapCompat.getOrDefault(kbdMap, KMManager.KMKey_KMPLink, ""),
-              isNewKeyboard,
-              MapCompat.getOrDefault(kbdMap, KMManager.KMKey_Font, null),
-              MapCompat.getOrDefault(kbdMap, KMManager.KMKey_OskFont, null)
-            );
-            list.add(k);
-          }
-
+          KMManager.updateOldKeyboardsList(context, dat_list, list);
```

Comment:
Again, why are we passing `list` in as a parameter?  Examining the code above, we know that its value is guaranteed to be an empty list instance; `updateOldKeyboardsList` neither needs nor cares for the parameter's value except as an output.